src/datepicker/datepicker.ts

```diff
@@ -298,6 +329,29 @@ export class NgbDatepicker implements OnDestroy,
     });
   }
 
+  /**
+   *  Returns a copy of the state of the datepicker.
+   */
+  getState(): NgbDatepickerState {
+    return Object.create({
```

Comment:
1. We don't want to return a new object every time anybody calls `dp.getState()`. It might happen on every change detection, literally many times per second. You need to create a new object when `this.model` is updated only. Also there is no need in `Object.create({})`, just `{}` will do.
2. Let's change it to a getter → `get state(): NgbDatepickerState` as agreed ?